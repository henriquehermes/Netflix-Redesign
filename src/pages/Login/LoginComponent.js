import React, { useState } from 'react';

import { Container, Netflix, Body, Background } from './LoginStyles';
import Button from '~/components/Button';
import TextInput from '~/components/Input';

export default function Login({ loginInApp }) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  return (
    <Container>
      <Netflix />
      <Body>
        <TextInput
          placeholder="Email"
          value={email}
          onChangeText={text => setEmail(text)}
        />
        <TextInput
          placeholder="Password"
          value={password}
          onChangeText={text => setPassword(text)}
          type="password"
          secureTextEntry
        />
        <Button
          onPress={() => loginInApp(email, password)}
          label="Sign In"
          style={{ marginTop: 15 }}
        />
      </Body>
      <Background />
    </Container>
  );
}
