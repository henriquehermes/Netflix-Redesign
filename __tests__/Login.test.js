/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from 'react-native-testing-library';

import LoginComponent from '../src/pages/Login/LoginComponent';

const fakeIsLoading = false;
const fakeLoginInApp = jest.fn();

it('renders correctly', () => {
  const tree = renderer
    .create(
      <LoginComponent isLoading={fakeIsLoading} navigation={fakeLoginInApp} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('change email', () => {
  const { getByTestId } = render(
    <LoginComponent isLoading={fakeIsLoading} navigation={fakeLoginInApp} />,
  );

  const input = getByTestId('testEmail');

  const testText = 'teste@email.com';
  fireEvent(input, 'onChangeText', testText);
});

it('change password', () => {
  const { getByTestId } = render(
    <LoginComponent isLoading={fakeIsLoading} navigation={fakeLoginInApp} />,
  );

  const input = getByTestId('testPassword');

  const testText = '321';
  fireEvent(input, 'onChangeText', testText);
});
