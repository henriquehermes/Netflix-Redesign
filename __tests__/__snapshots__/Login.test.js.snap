// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#111111",
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": 20,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/img/Netflix_Logo.png",
      }
    }
    style={
      Array [
        Object {
          "height": 100,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 0.7,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="rgba(255,255,255,0.2)"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#1c1c1c",
            "borderColor": "#1c1c1c",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 2,
            "color": "rgba(255, 255, 255, 0.87)",
            "fontSize": 14,
            "marginBottom": 10,
            "maxWidth": 300,
            "paddingBottom": 5,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 5,
            "width": "100%",
          },
        ]
      }
      testID="testEmail"
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="rgba(255,255,255,0.2)"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "backgroundColor": "#1c1c1c",
            "borderColor": "#1c1c1c",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 2,
            "color": "rgba(255, 255, 255, 0.87)",
            "fontSize": 14,
            "marginBottom": 10,
            "maxWidth": 300,
            "paddingBottom": 5,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 5,
            "width": "100%",
          },
        ]
      }
      testID="testPassword"
      type="password"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#d7240f",
          "borderRadius": 4,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 15,
          "minHeight": 40,
          "minWidth": 100,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 12,
              "fontWeight": "600",
              "textTransform": "uppercase",
            },
          ]
        }
      >
        Sign In
      </Text>
    </View>
  </View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/img/Netflix_Background.jpg",
      }
    }
    style={
      Array [
        Object {
          "opacity": 0.15,
          "position": "absolute",
          "zIndex": -999,
        },
      ]
    }
  />
</View>
`;
